diff --git a/src/Lazy/LazyUuidFromString.php b/src/Lazy/LazyUuidFromString.php
index 8ba7579..ea79ef0 100644
--- a/src/Lazy/LazyUuidFromString.php
+++ b/src/Lazy/LazyUuidFromString.php
@@ -302,6 +302,7 @@ final class LazyUuidFromString implements UuidInterface
         return $this->uuid;
     }
 
+    #[\ReturnTypeWillChange]
     public function jsonSerialize(): string
     {
         return $this->uuid;
diff --git a/src/Type/Decimal.php b/src/Type/Decimal.php
index 10f9384..e940b39 100644
--- a/src/Type/Decimal.php
+++ b/src/Type/Decimal.php
@@ -90,6 +90,7 @@ final class Decimal implements NumberInterface
         return $this->toString();
     }
 
+    #[\ReturnTypeWillChange]
     public function jsonSerialize(): string
     {
         return $this->toString();
diff --git a/src/Type/Hexadecimal.php b/src/Type/Hexadecimal.php
index 88adc2e..4cf19d3 100644
--- a/src/Type/Hexadecimal.php
+++ b/src/Type/Hexadecimal.php
@@ -69,6 +69,7 @@ final class Hexadecimal implements TypeInterface
         return $this->toString();
     }
 
+    #[\ReturnTypeWillChange]
     public function jsonSerialize(): string
     {
         return $this->toString();
diff --git a/src/Type/Integer.php b/src/Type/Integer.php
index 7690f6c..86e4c71 100644
--- a/src/Type/Integer.php
+++ b/src/Type/Integer.php
@@ -106,6 +106,7 @@ final class Integer implements NumberInterface
         return $this->toString();
     }
 
+    #[\ReturnTypeWillChange]
     public function jsonSerialize(): string
     {
         return $this->toString();
diff --git a/src/Type/Time.php b/src/Type/Time.php
index dd1b8bc..97b7c51 100644
--- a/src/Type/Time.php
+++ b/src/Type/Time.php
@@ -77,6 +77,7 @@ final class Time implements TypeInterface
     /**
      * @return string[]
      */
+    #[\ReturnTypeWillChange]
     public function jsonSerialize(): array
     {
         return [
diff --git a/src/Uuid.php b/src/Uuid.php
index 945480b..9576511 100644
--- a/src/Uuid.php
+++ b/src/Uuid.php
@@ -278,6 +278,7 @@ class Uuid implements UuidInterface
     /**
      * Converts the UUID to a string for JSON serialization
      */
+    #[\ReturnTypeWillChange]
     public function jsonSerialize(): string
     {
         return $this->toString();
